applications/StructuralMechanicsApplication/tests/response_function_tests/adjoint_stress_response_parameters.json

```diff
@@ -13,6 +13,7 @@
         "response_type"    : "adjoint_local_stress",
         "gradient_mode"     : "semi_analytic",
         "step_size"         : 1.0e-5,
+        "adapt_step_size"  : true,
```

Comment:
The step_size/perturbation size difference is already there.
I would vote for `adapt_step_size"` in the json settings and `ADAPT_PERTURBATION_SIZE` for the variable name.
If the json settings are changed from `"step_size"` to `"perturbation_size"`, it will change many existing examples...